nengo/utils/graphs.py

```diff
@@ -98,6 +98,43 @@ def toposort(edges):
     return ordered
 
 
+def transitive_closure(edges, topo_sorted=None):
+    """Constructs the transitive_closure of a DAG.
```

Comment:
I think “transitive closure” is a well-defined term and can be easily looked up (e.g. on Wikipedia). Technically, it is not a reach-ability *matrix* here because the graph is not represented with a adjancency matrix, but something more similar to adjancency lists.